File: code/data_structures/queue/reverse_queue/reverse_queue_.cpp
// CPP program to reverse a Queue
#include <bits/stdc++.h>
using namespace std;
 
// Utility function to print the queue
void Print(queue<int>& Queue)
{
    int temp;
    queue<int> Queue2;
    while (!Queue.empty()) {
        temp = Queue.front();
        cout << Queue.front() << " ";
        Queue.pop();
        Queue2.push(temp);
    }
    while (!Queue2.empty()) {
        Queue.push(Queue2.front());
        Queue2.pop();
    }
}
 
// Function to reverse the queue
void reverseQueue(queue<int>& Queue)
{
    stack<int> Stack;
    while (!Queue.empty()) {
        Stack.push(Queue.front());
        Queue.pop();
    }
    while (!Stack.empty()) {
        Queue.push(Stack.top());
        Stack.pop();
    }
}
 
// Driver code
int main()
{
    queue<int> Queue;
    int exit = 0;
    int input;
    char entry;
    while(!exit)
    {
        cout<<"Enter p to push into queue,r to reverse the queue,s to show the queue,x to exit: ";
        cin>>entry;
        if(entry == 'p')
        {
            cout<< "Enter number to push: ";
            cin>>input;
            Queue.push(input);
        }
        if(entry=='r')
            reverseQueue(Queue);
        
        if(entry=='s')
        {
            Print(Queue);
            cout<<endl;
        }
        if(entry=='x')
            exit=1;
    }
    
    return 0;
}

File: code/data_structures/src/queue/reverse_queue/reverse_queue.cs
// c# program to reverse a Queue  
using System; 
using System.Collections.Generic; 
  
public class RONALDO 
{ 
  
public static LinkedList<int> queue; 
  
// Utility function to print the queue  
public static void Print() 
{ 
    while (queue.Count > 0) 
    { 
        Console.Write(queue.First.Value + ", "); 
        queue.RemoveFirst(); 
    } 
} 
  
// Function to reverse the queue  
public static void reversequeue() 
{ 
    Stack<int> stack = new Stack<int>(); 
    while (queue.Count > 0) 
    { 
        stack.Push(queue.First.Value); 
        queue.RemoveFirst(); 
    } 
    while (stack.Count > 0) 
    { 
        queue.AddLast(stack.Peek()); 
        stack.Pop(); 
    } 
} 
  
// Driver code  
public static void Main(string[] args) 
{ 
    queue = new LinkedList<int>(); 
    queue.AddLast(10); 
    queue.AddLast(20); 
    queue.AddLast(30); 
    queue.AddLast(40); 
    queue.AddLast(50); 
    queue.AddLast(60); 
    queue.AddLast(70); 
    queue.AddLast(80); 
    queue.AddLast(90); 
    queue.AddLast(100); 
  
    reversequeue(); 
    Print(); 
} 
} 
